src/adapters/bam/llm.ts

```diff
@@ -221,6 +229,30 @@ export class BAMLLM extends LLM<BAMLLMOutput, BAMLLMGenerateOptions> {
     }
   }
 
+  async embed(input: LLMInput[], options?: EmbeddingOptions): Promise<EmbeddingOutput> {
+    const results = await Promise.all(
+      chunk(input, MAX_EMBEDDING_INPUTS).map((texts) =>
+        BAMLLM.limit(async () => {
+          const response = await this.client.text.embedding.create(
+            {
+              model_id: this.modelId,
+              input: texts,
+              parameters: {
+                truncate_input_tokens: true,
+              },
+            },
+            { signal: options?.signal },
+          );
+          if (response.results?.length !== texts.length) {
+            throw new Error("Missing embedding");
+          }
+          return response.results.map((result) => result.embedding);
+        }),
+      ),
+    );
+    return { embeddings: results.flat() };
+  }
```

Comment:
BAM SDK uses a concurrency limiter internally; do not worry about it.

Reply:
I removed the limiter